### PR TITLE
Added data strutures and a section on Controllers.

### PR DESCRIPTION
**Pull request by degregat**

### CODE REVIEW

File: src/architecture/layers/resources.md [hunk 1]
```diff
@@ -59,29 +71,77 @@ If we want to upgrade the proof system used for a Resource type (determined by i
...
+
+Examples of optional data:
 - Any additional static data that is required, e.g. External Identities.
 
-### quantity
-An integer valued quantity, to determine balance in `tx`s using fungible `Resources`. **Note: We might want to move this into resource_data_dynamic.**
+```haskell=
+data ResourceDataStatic = ResourceDataStatic {
+  prefix :: [ContentHash],
+  proof_system :: ByteString, TODO: Taiga: Is this a good name and datatype?
```
Comment (degregat) [inline src/architecture/layers/resources.md:113]: @joebebel can you have a look at this?

File: src/architecture/layers/resources.md [hunk 2]
```diff
@@ -24,25 +24,37 @@ The Resource Logic is tied to a specific Proof System for shielded `ptx`s, since
...
 
 ### Transaction (tx)
 Transactions provide the notion of balance for a set of `ptx`s, as well as validity for all their Predicates.
 
 ### Proof System
 The Proof System of a resource is defined via the proof and functional commitment schemes used. It is encoded (amongst other things relevant to the resource logic, e.g. public keys etc.) statically in `resource_data_static` to increase legibility for differences between Resource Logics, by separating Predicate from Proof System differences.
 
+### Controllers
+Controllers are the Identities (e.g. Consensus Providers or the Resource Originator) that determine the order of (p)txs including the given Resource.
+By signing a message, a Controller promises to not sign another message implementing an equivocation of the signed message.
```
Comment (cwgoes) [inline src/architecture/layers/resources.md:48]:
```suggestion
By signing a message, a Controller promises to not sign another message committing to an equivocation of the signed message. 
```

File: src/architecture/layers/resources.md [hunk 3]
```diff
@@ -59,29 +71,77 @@ If we want to upgrade the proof system used for a Resource type (determined by i
...
+```haskell=
+data ResourceLogic = ResourceLogic {
+  creationPredicate :: TxData -> Bool,
+  consumptionPredicate :: TxData -> Bool,
+}
+```
+
 ### resource_data_static
-This field contains everything that is relevant to fungibility of the _Resource_, which is not a Predicate, such as:
+This struct contains everything that is relevant to fungibility of the _Resource_, which is not a Predicate, such as.
```
Comment (cwgoes) [inline src/architecture/layers/resources.md:99]: such as what?

File: src/architecture/layers/resources.md [hunk 4]
```diff
@@ -59,29 +71,77 @@ If we want to upgrade the proof system used for a Resource type (determined by i
...
 ### resource_data_dynamic
-This field contains everything which _is not_ relevant to the fungibility of the _Resource_, including Predicates:
-- The `Value` of the encoded `Resource` represented by a bytestring (i.e. the current content of the Memory behind the address).
+This struct contains everything which _is not_ relevant to the fungibility of the _Resource_, including Predicates.
+
+Mandatory fields:
 - The dynamic `Suffix` identifying the version of the `Value` of the Resource.
+- The list of `Controllers`for the Resource.
+- The `Value` of the encoded `Resource` represented by a ByteString (i.e. the current content of the Memory behind the address).
+- An integer valued quantity, to determine balance in `tx`s using fungible `Resources`.
```
Comment (cwgoes) [inline src/architecture/layers/resources.md:126]:
```suggestion
- An integer valued `Quantity`, to determine balance in `tx`s using fungible `Resources`.
```

**Review by cwgoes — CHANGES_REQUESTED** [review]

**Comment by cwgoes** [comment]
TODO: Go over the controller non-equivocation logic in detail & with Issac.

File: src/architecture/layers/resources.md [hunk 5]
```diff
@@ -59,29 +71,77 @@ If we want to upgrade the proof system used for a Resource type (determined by i
...
+  resource_data_dynamic :: ResourceDataDynamic,
+}
+```
+    
 ### resource_logic
 This field contains all Predicate components that are relevant to the fungibility of a Resource it encodes. It can optionally call external predicates which are stored in `resource_data_dynamic` or other `Resource`s.
 
+```haskell=
+data ResourceLogic = ResourceLogic {
+  creationPredicate :: TxData -> Bool,
```
Comment (degregat) [inline src/architecture/layers/resources.md:93]:
```suggestion
   predicate :: TxData -> Bool,
```

File: src/architecture/layers/resources.md [hunk 6]
```diff
@@ -59,29 +71,77 @@ If we want to upgrade the proof system used for a Resource type (determined by i
...
+  consumptionPredicate :: TxData -> Bool,
```
Comment (degregat) [inline src/architecture/layers/resources.md:94]:
```suggestion
```

File: src/architecture/layers/resources.md [hunk 7]
```diff
@@ -24,25 +24,37 @@ The Resource Logic is tied to a specific Proof System for shielded `ptx`s, since
...
+By signing a message, a Controller promises to not sign another message implementing an equivocation of the signed message. 
+It is recommended to assume finality only after checking Controller Signatures on (p)txs.
+
+Precedence of Controller signatures is ordered by their position in the list, i.e. a signature from the first Controller can override signatures of all downstream Controllers.
+Because of this, the list positions should correspond to Trust, from most trusted in the front, to least trusted at the end.
+
+This way we gain the following options by using signatures of upstream Controllers:
+- Resolving conflicts created by defecting Controllers.
+- Updating the Controller list, when the most downstream Controller is offline or defected.
+
```
Comment (degregat) [inline src/architecture/layers/resources.md:57]:
```suggestion

> TODO: Revise and concretise this section.
```

**Pull request merged**